src/Site.js

```diff
@@ -306,7 +306,7 @@ Site.prototype.createPage = function (config) {
     faviconUrl: config.faviconUrl,
     pageTemplate: this.pageTemplate,
     rootPath: this.rootPath,
-    searchable: config.searchable,
+    searchable: (!('enableSearch' in this.siteConfig) || this.siteConfig.enableSearch) && config.searchable,
```

Comment:
Possible to just set the variable during `readSiteConfig()`, so that it is simplified to `searchable: this.siteConfig.enableSearch && config.searchable`?

```diff
diff --git a/src/Site.js b/src/Site.js
index 3373f29..2e41cb2 100644
--- a/src/Site.js
+++ b/src/Site.js
@@ -274,6 +274,9 @@ Site.prototype.readSiteConfig = function (baseUrl) {
       .then((config) => {
         this.siteConfig = config;
         this.siteConfig.baseUrl = (baseUrl === undefined) ? this.siteConfig.baseUrl : baseUrl;
+        if (this.siteConfig.enableSearch === undefined) {
+          this.siteConfig.enableSearch = true;
+        }
         resolve(this.siteConfig);
       })
       .catch((err) => {
```